### PR TITLE
added assetName. marked plugin readonly. added ordering.

### PR DESCRIPTION
**Pull request by billehunt**
```
collected 6 items                                                                                                                             

test/test_sinusoid.py::test_plugin_contract PASSED
test/test_sinusoid.py::test_plugin_info PASSED
test/test_sinusoid.py::test_plugin_init PASSED
test/test_sinusoid.py::test_plugin_start SKIPPED
test/test_sinusoid.py::test_plugin_reconfigure SKIPPED
test/test_sinusoid.py::test_plugin_shutdown PASSED

== 4 passed, 2 skipped in 0.05 seconds =
```

### CODE REVIEW

File: python/foglamp/plugins/south/sinusoid/sinusoid.py [hunk 1]
```diff
@@ -25,14 +25,22 @@
...
 _DEFAULT_CONFIG = {
     'plugin': {
-        'description': 'Sinusoid async plugin',
+        'description': 'Sinusoid Plugin',
         'type': 'string',
-        'default': 'sinusoid'
+        'default': 'sinusoid',
+        'readonly': 'true'
+    },
+    'assetName': {
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/sinusoid/sinusoid.py:32]: make sure to have it in plugin_reconfigure as`if 'dataPointsPerSec' in diff or 'assetName' in diff:` 

```
# Plugin should re-initialize and restart if key configuration is changed
if 'dataPointsPerSec' in diff:
```

**Comment by billehunt** [comment]
Sounds good. @Ashish, please add that.

**Review by Singhal-Vaibhav — APPROVED** [review]
Looks good. All points addressed and on reconfig plugin restart

**Review by praveen-garg — APPROVED** [review]

**Comment by praveen-garg** [comment]
Well! go ahead. but as we discussed this will not work for released 1.3.1 but latest develop only.

**Pull request merged**